tests/claws-utils/box-test.cpp

```diff
@@ -0,0 +1,39 @@
+#include <type_traits>
+#include <gtest/gtest.h>
+#include <claws/utils/box.hpp>
+
+TEST(box, type_traits)
+{
+  using int_box = claws::box<int>;
+  static_assert(std::is_class_v<int_box>);
+  static_assert(std::is_same_v<typename int_box::type, int>);
+}
+
+TEST(box, constexpr_correctness)
+{
+  constexpr claws::box<int> i{2};
+  static_assert(i == 2);
+  constexpr claws::box<int> i2 = i;
+  static_assert(i2 == 2);
+  constexpr claws::box<int> i3 = std::move(i);
+  static_assert(i3 == 2);
+  constexpr int value = i3;
+  static_assert(value == 2);
+}
+
+TEST(box, noexcept_correctness)
+{
+  using int_box = claws::box<int>;
+  static_assert(noexcept(int_box()));
+  static_assert(noexcept(int_box(std::declval<const int_box &>())));
+  static_assert(noexcept(int_box(std::declval<int_box &&>())));
+
+  static_assert(noexcept(std::declval<int_box &>() = std::declval<const int_box &>()));
+  static_assert(noexcept(std::declval<int_box &>() = std::declval<int_box &&>()));
+
+  static_assert(noexcept(int_box(std::declval<const int &>())));
+  static_assert(noexcept(int_box(std::declval<int &&>())));
+
+  static_assert(noexcept(static_cast<const int &>(std::declval<const int_box &>())));
+  static_assert(noexcept(static_cast<int &>(std::declval<int_box &>())));
```

Comment:
Would be there a way to make a negative test? ex: check that claws::box doesn't compile with a class.

Reply:
I don't see any way to achieve that here, as the static assertion in `claws::box` is a definition error, so SFINAE cannot be used to recover from it.

However, we could provide a `claws::is_boxable` type trait and change `claws::box` to use it, then test that trait.

Reply:
I was wondering if such a thing would be present in google test. A trait isn't a bad idea, I'll open an issue.